added synthetic benchmarks

arberzela opened the pull request.

Commit:
added synthetic benchmarks

Commit:
copied nos_space here temporarily for convenience

Niccolo-Ajroldi commented:
Looks great thanks! Can you just remove image files from the Git index for now? So that we keep it clean. Thanks!

arberzela committed:
added baselines and some more plotting functionality

arberzela committed:
Merge branch 'Niccolo-Ajroldi:opt_search' into opt_search

arberzela committed:
removed figures

arberzela commented:
@Niccolo-Ajroldi thanks for checking it. I also added some baseline optimizers (Adam, SGD) and added a plotting script to visualize the optimization trajectory in 2D. Removed the image files as well as you suggested :)

arberzela committed:
added more benchmarks and regularized evolution

arberzela committed:
updated search space

arberzela committed:
removed noisy linear regression MSE loss from benchmarks
